File: cyanBat/src/main/java/at/smiech/cyanbat/screens/CyanBatBaseScreen.java
package at.smiech.cyanbat.screens;

import android.graphics.Color;

import at.grueneis.game.framework.Game;
import at.grueneis.game.framework.Graphics;
import at.grueneis.game.framework.Screen;

public abstract class CyanBatBaseScreen extends Screen {
    public static int DISPLAY_HEIGHT = 480;
    public static int DISPLAY_WIDTH = 320;

    public static Game game;

    public CyanBatBaseScreen(Game game) {
        super(game);
        CyanBatBaseScreen.game = game;
    }


    public void drawMap(Graphics g) {
        g.clear(Color.BLACK);
    }


    // Default screen implementations - override if needed
    @Override
    public void update(float deltaTime) {
        // empty default method
    }

    @Override
    public void present(float deltaTime) {
        // empty default method
    }

    @Override
    public void pause() {
        // empty default method

    }

    @Override
    public void resume() {
        // empty default method

    }

    @Override
    public void dispose() {
        // empty default method

    }

}

File: cyanBat/src/main/java/at/smiech/cyanbat/screens/CyanBatBaseScreen.kt
package at.smiech.cyanbat.screens

import android.graphics.Color

import at.grueneis.game.framework.Game
import at.grueneis.game.framework.Graphics
import at.grueneis.game.framework.Screen

abstract class CyanBatBaseScreen(game: Game) : Screen(game) {



    fun drawMap(g: Graphics) {
        g.clear(Color.BLACK)
    }


    // Default screen implementations - override if needed
    override fun update(deltaTime: Float) {
        // empty default method
    }

    override fun present(deltaTime: Float) {
        // empty default method
    }

    override fun pause() {
        // empty default method

    }

    override fun resume() {
        // empty default method

    }

    override fun dispose() {
        // empty default method

    }

    companion object {
        var DISPLAY_HEIGHT = 480
        var DISPLAY_WIDTH = 320
    }

}
